src/main/java/twilightforest/entity/boss/EntityTFNaga.java

```diff
@@ -777,6 +783,8 @@ public boolean attackEntityFromPart(EntityDragonPart entitydragonpart, DamageSou
     @Override
     public Entity[] getParts()
     {
-        return body;
+    	List<EntityTFNagaSegment> list = new ArrayList<EntityTFNagaSegment>(Arrays.asList(body));
+    	list.removeAll(Collections.singleton(null));
+    	return list.toArray(new EntityTFNagaSegment[list.size()]); // This cannot contain nulls
```

Comment:
So I actually have very little to no experience with Java 8 streams, and this and casting wont quite work, what would be the proper way short way to get to this return as Entity[]?